app/validators/url_validator.rb

```diff
@@ -0,0 +1,29 @@
+class UrlValidator < ActiveModel::EachValidator
+  InvalidSchemeError = Class.new StandardError
+  MissingHostError = Class.new StandardError
+
+  DEFAULT_SCHEMES = %w[http https].freeze
+
+  def validate_each(record, attribute, value)
+    uri = URI.parse(value)
+    validate_scheme uri
+    validate_host uri
+  rescue URI::InvalidURIError
+    record.errors.add(attribute)
```

Comment:
We probably want to add a message that let's the user know the URL they entered is not formatted correctly

Reply:
Since we can't know the exact issue, I just added a generic invalid format message.